src/services/query.js

```diff
@@ -308,30 +323,36 @@ ngeo.Query.prototype.issueWMSGetFeatureInfoRequests_ = function(
       return;
     }
 
-    // skip layers that don't have a source configured
+    // skip layers that don't have one or more sources configured
     id = this.getLayerSourceId_(layer);
-    if (!id || !this.cache_[id]) {
+    ids = this.getLayerSourceIds_(layer);
+    if ((!id || !this.cache_[id]) && !ids.length) {
       return;
     }
 
-    item = this.cache_[id];
-    item['resultSource'].pending = true;
-    infoFormat = item.source.infoFormat;
-
-    // sources that use GML as info format are combined together if they
-    // share the same server url
-    if (infoFormat === ngeo.QueryInfoFormatType.GML) {
-      url = item.source.wmsSource.getUrl();
-      goog.asserts.assertString(url);
-      if (!itemsByUrl[url]) {
-        itemsByUrl[url] = [];
-      }
-      itemsByUrl[url].push(item);
-    } else {
-      // TODO - support other kinds of infoFormats
-      item['resultSource'].pending = false;
+    if (id) {
+      ids.push(id);
```

Comment:
`ids` is the array stored on the layer? If `issueWMSGetFeatureInfoRequests` is called multiple times, you also add `id` multiple times as you are not working on a clone of the array?